src/kartograf/atom_mapper.py

```diff
@@ -855,6 +854,20 @@ def suggest_mapping_from_rdmols(
 
         return mapping
 
+    def _split_protein_component_chains(self, protein: ProteinComponent)->list[ProteinComponent]:
+        chain_components = []
+        rdmol = protein.to_rdkit()
+        # TODO: Implement splitting up A into chain components
+
+        return chain_components
+
+    def _merge_protein_component_chain_mappings(self, A:ProteinComponent, B:ProteinComponent,
+                                                sub_mappings: list[AtomMapping])->AtomMapping:
+        merged_mapping = {}
+        # TODO: Implement finding largest mapped overlaps in all possible chains
+        # TODO: return merged mapping
+
+        return AtomMapping(componentA=A, componentB=B, mapping=merged_mapping)
```

Comment:
@ijpulidos thx for bringing up this issue!
here I added some pseudo code stubs, that explain what I think needs to be implemented in order to solve your problem! :)